From 303ff4b8e0235e7501776c15089effb16e02d904 Mon Sep 17 00:00:00 2001
From: Hugh McMaster <hugh.mcmaster@outlook.com>
Date: Thu, 29 Oct 2015 23:17:17 +1100
Subject: wineconsole: Add if check to determine whether a font attribute has
 changed (v3)

This version uses a case insensitive comparison, as requested by Dmitry Timoshkov.

Patches two to six are unchanged.

Signed-off-by: Hugh McMaster <hugh.mcmaster@outlook.com>
---
 programs/wineconsole/wineconsole.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/programs/wineconsole/wineconsole.c b/programs/wineconsole/wineconsole.c
index e1ee55b..eb07f6b 100644
--- a/programs/wineconsole/wineconsole.c
+++ b/programs/wineconsole/wineconsole.c
@@ -27,7 +27,7 @@
 #include "winecon_private.h"
 #include "winnls.h"
 #include "winuser.h"
-
+#include "wine/unicode.h"
 #include "wine/debug.h"
 
 WINE_DEFAULT_DEBUG_CHANNEL(wineconsole);
@@ -421,7 +421,8 @@ void     WINECON_SetConfig(struct inner_data* data, const struct config_data* cf
     }
     data->curcfg.menu_mask = cfg->menu_mask;
     data->curcfg.quick_edit = cfg->quick_edit;
-    if (1 /* FIXME: font info has changed */)
+    if (strcmpiW(data->curcfg.face_name, cfg->face_name) || data->curcfg.cell_width != cfg->cell_width ||
+        data->curcfg.cell_height != cfg->cell_height || data->curcfg.font_weight != cfg->font_weight)
     {
         data->fnSetFont(data, cfg->face_name, cfg->cell_height, cfg->font_weight);
     }
-- 
2.6.1

